### ISSUE TITLE
Clang-tidy: False positive readability-redundant-inline-specifier

### ISSUE THREAD

**Issue by joto** [llvm/llvm-project#122291]
clang-tidy shows "readability-redundant-inline-specifier" messages for methods defined outside the class in a header file. But it should not do that, because it violates the one-definition-rule. Usually those definitions would be in a cpp file not a header file, but sometimes they are not, in a header-only library for instance.

**Comment by Nekto89** [comment]
> Can you please provide a small example to reproduce?

clang-tidy 19 from Debian Bookworm (https://packages.debian.org/bookworm/clang-tools-19) removes inline from destructor in header files that leads to multiple definitions afterwards. I don't know whether it is still the issue in the trunk.

```
#pragma once

struct A {
  ~A();
};

inline A::~A() = default;
```

**Comment by EugeneZelenko** [comment]
@Nekto89: You could try `main` on https://godbolt.org.

**Comment by localspook** [comment]
#130745 looks like a duplicate of this issue, and that one led to a fix, so I think this issue is resolved.
